agent/s3/factory/factory.go

```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/credentials"
 	"github.com/aws/amazon-ecs-agent/agent/httpclient"
 	s3client "github.com/aws/amazon-ecs-agent/agent/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
```

Comment:
nit: i think this import needs to be clubbed in the block below, with other external imports